Mathlib/Mathport/Syntax.lean

```diff
@@ -0,0 +1,727 @@
+syntax (name := deriveElementwiseProof) "deriveElementwiseProof" : tactic
+
+syntax (name := mkDecorations) "mkDecorations" : tactic
+
+syntax (name := nontriviality) "nontriviality"
+  (ppSpace (colGt term))? (" using " simpArg,+)? : tactic
+
+syntax (name := filterUpwards) "filterUpwards" " [" term,* "]" (ppSpace (colGt term))? : tactic
+
+syntax (name := isBounded_default) "isBounded_default" : tactic
+
+syntax (name := opInduction) "opInduction" (ppSpace (colGt term))? : tactic
+
+syntax (name := mvBisim) "mvBisim" (ppSpace (colGt term))? (" with " binderIdent+)? : tactic
+
+syntax (name := continuity) "continuity" (" (" &"config" " := " term ")")? : tactic
+syntax (name := continuity!) "continuity!" (" (" &"config" " := " term ")")? : tactic
+syntax (name := continuity?) "continuity?" (" (" &"config" " := " term ")")? : tactic
+syntax (name := continuity!?) "continuity!?" (" (" &"config" " := " term ")")? : tactic
+
+syntax (name := unitInterval) "unitInterval" : tactic
+syntax (name := mfldSetTac) "mfldSetTac" : tactic
+
+syntax (name := measurability) "measurability" (" (" &"config" " := " term ")")? : tactic
+syntax (name := measurability!) "measurability!" (" (" &"config" " := " term ")")? : tactic
+syntax (name := measurability?) "measurability?" (" (" &"config" " := " term ")")? : tactic
+syntax (name := measurability!?) "measurability!?" (" (" &"config" " := " term ")")? : tactic
+syntax (name := padicIndexSimp) "padicIndexSimp" " [" term,* "]" (ppSpace location)? : tactic
+
+syntax (name := uniqueDiffWithinAt_Ici_Iic_univ) "uniqueDiffWithinAt_Ici_Iic_univ" : tactic
+
+syntax (name := ghostFunTac) "ghostFunTac " term ", " term : tactic
+syntax (name := ghostCalc) "ghostCalc" (ppSpace binderIdent)* : tactic
+syntax (name := initRing) "initRing" (" using " term)? : tactic
+syntax (name := ghostSimp) "ghostSimp" (" [" simpArg,* "]")? : tactic
+syntax (name := wittTruncateFunTac) "wittTruncateFunTac" : tactic
+
+namespace Conv
+
+syntax (name := convConv) "conv" " => " Conv.convSeq : conv
+syntax (name := erw) "erw " rwRuleSeq : conv
+syntax (name := applyCongr) "applyCongr" (ppSpace (colGt term))? : conv
+syntax (name := guardTarget) "guardTarget" " =ₐ " term : conv
+
+syntax (name := normCast) "normCast" : conv
+
+syntax (name := normNum1) "normNum1" : conv
+syntax (name := normNum) "normNum" (" [" simpArg,* "]")? : conv
+
+syntax (name := ringNF) "ringNF" (ppSpace ringMode)? : conv
+syntax (name := ringNF!) "ringNF!" (ppSpace ringMode)? : conv
+syntax (name := ring) "ring" : conv
+syntax (name := ring!) "ring!" : conv
+
+syntax (name := ringExp) "ringExp" : conv
+syntax (name := ringExp!) "ringExp!" : conv
+
+syntax (name := slice) "slice " num num : conv
+
+end Conv
+end Tactic
+
+namespace Attr
+
+syntax (name := intro) "intro" : attr
+syntax (name := intro!) "intro!" : attr
+
+syntax (name := nolint) "nolint" (ppSpace ident)* : attr
+
+syntax (name := ext) "ext" (ppSpace ident)? : tactic
```

Comment:
This is not a tactic.  Also the ext attribute is never used with an argument in mathlib.
```suggestion
```

Reply:
```suggestion
syntax (name := ext) "ext" (ppSpace ident)? : attr
```
I know it's not used with a name right now, but this is part of the original tactic. Note that it is already simplified from the original syntax which used a list of names and `(->)`, this was PR leanprover-community/mathlib#8785 . But the name argument is actually useful in principle and I could not find a reason to drop the feature, although porting will be trivial since the feature isn't used.

In other words, this is a mathlib4 todo list for us to design the lean 4 tactic language we want to support with attention to lean 3 parity, not just a basis for synport. Removing the argument and erroring on any use of the with-argument form in synport is an option, but I don't want such decreases in functionality to sneak up on us without at least a quorum of some kind (ideally including the original author of the tactic, who likely put the feature in for a reason).

Reply:
The reason for the extra argument is afaict that the Lean 3 version of `ext` indexes the lemmas by the constant name of the head symbol.  The Lean 4 version uses discrimination trees for indexing and doesn't support this annotation.  (And it shouldn't, the lemma indexing and lookup done by the old ext tactic is a huge hack.)